### ISSUE TITLE
bug(Typography):  @include mat.typography-hierarchy(mat.define-typography-config()) sets the default font settings on `.mat-typography` to `body-2` instead of `body-1`

### ISSUE THREAD

**Issue by avidan-chen** [angular/components#27112]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

14

### Description

When a theme is configured according to the docs, and typography is set up by adding this line:
`@include mat.typography-hierarchy(mat.define-typography-config())`

When applying the `.mat-typography` class on the body element, the font-size that's being applied is body-2:
https://github.com/angular/components/blob/16.0.1/src/material/core/typography/_typography.scss#L273

body-1 has a font-size of 16px.
body-2 has a font-size of 14px;

The docs say:
 `body-1`         | Base body text.  
`body-2`         | Secondary body text.   
https://github.com/angular/components/blob/16.0.1/guides/typography.md?plain=1#L54

If `body-1` is considered `base body text`, I expect it to apply when using the `mat-typography` css class, and not `bode-2`.

### Reproduction

1. Follow the docs to set up typography styles in a brand new Angular + Angular Material app: 
https://v15.material.angular.io/guide/typography#using-typography-styles-in-your-application

2. Add `mat-typography` to the `body` tag.

3. Inspect font-size on the `body` tag. 

### Expected Behavior

I expect `body-1` to be the default font settings which apply when using `mat-typography` (since the docs claim this is `base body text`).

### Actual Behavior

`body-2` is the default font settings when using `@include mat.typography-hierarchy(mat.define-typography-config())` instead of `body-1`.

### Environment

- Angular: 15.2.9
- CDK/Material: 15.2.9
- Browser(s): Chrome / Edge
- Operating System (e.g. Windows, macOS, Ubuntu): macOS / Windows

**Comment by michaelfaith** [comment]
Later in the docs, though, the this table shows `body-2` as the standard body text, which is accurate.  So, is it just the table higher up that needs updating to be consistent? 

![image](https://github.com/angular/components/assets/8071845/33828e09-8cdc-4a04-b024-fed97a4ae67d)

**Comment by avidan-chen** [comment]
@michaelfaith in pre-mdc typography styles, the base body text, the default font was body-1:
https://github.com/angular/components/blob/14.2.7/src/material/core/typography/_typography.scss#L268

However, the definition for body-1 was the same as for body-2 (14px) for some reason (which seems counter intuitive to me, as I'd expect body-1 to be bigger than body-2:
https://github.com/angular/components/blob/14.2.7/src/material/core/typography/_typography.scss#L60

Also, in the Material specs it is clearly stated body-1 should be 16px:
https://m2.material.io/design/typography/the-type-system.html#type-scale
![Screenshot 2023-05-31 at 9 18 31](https://github.com/angular/components/assets/1239344/b3564822-4869-40d7-a4f8-5f2bd0b202e5)

With the new Material spec (v3), there are 3 font sizes for body, largest being 16px as well:
https://m3.material.io/styles/typography/type-scale-tokens
![Screenshot 2023-05-31 at 9 20 44](https://github.com/angular/components/assets/1239344/67a09bcd-121f-4035-8f1a-c72742b524c0)

Therefore I think the correct behavior would be that body-1 is set to 16px and not 14px.

**Comment by michaelfaith** [comment]
> Therefore I think the correct behavior would be that body-1 is set to 16px and not 14px.

Right, so it sounds like you agree with me then, since `body-1` *is* already set to 16px and `body-2` is 14px.  It's just that body-2 in MDC-based Angular Material is default, which means the size is consistent with pre-MDC.  If you they to change `body-1` (16px) to be default, then that would increase everyone's default font size to 16 and likely break a lot of people.  Keeping body-2 (14) as default not only makes the size consistent with pre-MDC styles, but it seems to be consistent with M3 styles, since you probably want the default to be "medium".  It seems like the easiest, least impactful, and most correct thing to do would be to just change the table you referred to to match the table lower down and leave body-2 (14px) as default. When they roll out M3-bases styling those tables will need to be overhauled again anyway.

**Comment by michaelfaith** [comment]
> > It's just that body-2 in MDC-based Angular Material is default
> 
> which is incorrect according to the specs, as I've demonstrated in my previous comment.


How is that incorrect?  The spec doesn't say which body should be default text size.  It just says body-1 is 16 px, which it is, and body-2 is 14px, which it is.  So it does match the spec. 


> > Keeping body-2 (14) as default not only makes the size consistent with pre-MDC styles, but it seems to be consistent with M3 styles, since you probably want the default to be "medium".
> 
> Incorrect IMO. See the links and screenshots from my previous comment.


How is that incorrect?  14px equates to medium body, which is what body-2 currently is, which is default.  How is that wrong?  I think you're just getting tripped up by the fact that it's named 2.

**Comment by avidan-chen** [comment]
> How is that incorrect? The spec doesn't say which body should be default text size.

My pre-assumption that default body text should be body-1 was because in pre-MDC, body-1 was the default (although it was set to 14px and not 16px as it should have been).

In [M2 spec](https://m2.material.io/design/typography/the-type-system.html#type-scale), it says:
> The default for modern web browsers is 16px

Taking these two into consideration, it makes more sense to me that the default behavior out of the box would be to set the font-size to 16px, but I agree, it's not clearly defined.

Aligning the docs would definitely improve the situation though.

**Comment by michaelfaith** [comment]
If nobody's taken it up by the weekend, I can take a stab at aligning the docs.

**Comment by michaelfaith** [comment]
@avidan-chen I had time to put a PR in for this, if you want to take a look and see what you think

**Comment by wshager** [comment]
We had the same problem, so I'd like to keep this issue active. I think this is the most confusing part:

```
| `body-1`         | Base body text.           |
| `body-2`         | Secondary body text. |
```

Also note that form controls have body-1 styles, while paragraphs and such have body-2 styles. This clearly differs from Angular < 15, where both had body-1 styles. So confusing and inconsistent.

**Comment by marek-aguita** [comment]
I think the problem is in material/core/typography/_typography.scss, where body-2 typography is applied in .mat-typography class: 
![image](https://github.com/angular/components/assets/15703819/b258f89e-e150-476c-8c75-b564dda07028)

In my opinion correct would be to just change body-2 to body-1 in this scss code snippet.

**Comment by HakanSvenssonNexer** [comment]
I also ran into this issue and think is a bug.  
@marek-aguita do you know if your suggested solution will be implemented?

One "proof" of that this does not work as expected is this https://material.angular.io/guide/typography 

It says that it shall be styled as body-1 but it gets font 14px (and not 16 as body-1 should be)
![image](https://github.com/angular/components/assets/108413996/eeeb9659-9cc8-4039-8e1f-469ac3c55a18)

Another example: mat-card-content gives 16px and so that <p> also should get 16px seems logically correct (body-1)

**Comment by JoshuaWi** [comment]
Hi all, is there any movement on this issue? We are busy migrating from v14 - v15 and are now running into this issue where all our things that were previously set to 16px/1rem are now being changed to 14px/0.875rem because the main `mat-typography` that we are applying on the body tag now defaults to using `mat-body-2` instead of `mat-body-1` in v14
v14:
![image](https://github.com/angular/components/assets/50867148/d82caa46-ac3f-4e60-b3db-9ad95258e0e5)
v15:
![image](https://github.com/angular/components/assets/50867148/4eab3f10-517a-4fb0-811d-9b756998fe47)

**Comment by notbenj** [comment]
Hello,

I'm encoutering this issue on my project, and I also think mat-body should apply mat-body-1 as before instead of mat-body-2.

This is a breaking change, and I think it should be considered as a bug and fixed to the previous behavior.
